File: editorial_ch/chap5-10/prob5-10-2.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 輸入
		Scanner sc = new Scanner(System.in);
		long N = sc.nextLong();
		
		// 求出答案
		long mod = 1000000007;
		long val = N * (N + 1) / 2;
		val %= mod;
		System.out.println(val * val % mod);
	}
}

File: editorial_ch/chap5-10/prob5-10-2.cpp
#include <iostream>
using namespace std;
 
const long long mod = 1000000007;
long long N;
 
int main() {
	// 輸入
	cin >> N;
	
	// 求出答案
	long long val = N * (N + 1) / 2;
	val %= mod;
	cout << val * val % mod << endl;
	return 0;
}